dwave/optimization/include/dwave-optimization/array.hpp

```diff
@@ -880,6 +890,35 @@ class Array {
         return true;
     }
 
+    // Return a cached value if available, and otherwise return the value returned
+    // by ``func()``.
+    // If the ``cache.has_value()`` returns ``false``, then the cache is ignored
+    // entirely.
+    template <class T, class Func>
```

Comment:
decided against
```cpp
  template <class T, std::invocable Func>
```
?

Reply:
It's implicit to `std::invoke_result_t` so decided to leave it out.